daemon/content.go

```diff
@@ -26,5 +30,126 @@ func (daemon *Daemon) configureLocalContentStore() (content.Store, leases.Manage
 	}
 	md := metadata.NewDB(db, cs, nil)
 	daemon.mdDB = db
-	return md.ContentStore(), metadata.NewLeaseManager(md), nil
+	return namespacedContentProvider(md.ContentStore(), ns), namespacedLeaseManager(metadata.NewLeaseManager(md), ns), nil
+}
+
+func withNamespace(ctx context.Context, namespace string) context.Context {
```

Comment:
The function name (and lack of doc comment) does not make it clear that this sets the namespace on the context _iff one is not already set._
```suggestion
func withDefaultNamespace(ctx context.Context, namespace string) context.Context {
```